src/frontend/screens/SyncModal/UpgradeBar.js

```diff
@@ -0,0 +1,458 @@
+                  <View style={{ flex: 6, flexDirection: "column" }}>
+                    <Text style={styles.upgradeBarText} numberOfLines={2}>
+                      <FormattedMessage {...m.updateReady} />
+                    </Text>
+                  </View>
+                  <View style={{ flex: 2 }} />
+                  <View style={{ flex: 5, justifyContent: "center" }}>
+                    <TouchableNativeFeedback onPress={onInstallPress}>
+                      <TextButton
+                        title={t(m.installUpdateButton)}
+                        onPress={onInstallPress}
+                      />
+                    </TouchableNativeFeedback>
+                  </View>
+                </View>
+              );
+            case "Draining":
+              return (
+                <View
+                  style={{
+                    display: "flex",
+                    flexDirection: "row",
+                    flex: 1,
+                  }}
+                >
+                  <View style={{ flex: 6 }}>
+                    <Text style={styles.upgradeBarText} numberOfLines={2}>
+                      <FormattedMessage {...m.sharingUpdate} />
+                    </Text>
+                  </View>
+                  <View style={{ flex: 3 }} />
+                  <View style={{ flex: 1.5 }}>
+                    <MaterialIndicator
+                      animationDuration={8000}
+                      size={35}
+                      color="#3568f5"
+                    />
+                  </View>
+                </View>
+              );
+            default:
+              return (
+                <Text style={styles.upgradeBarText} numberOfLines={1}>
+                  <FormattedMessage {...m.updateUnkownState} />
+                </Text>
+              );
+          }
+        })()}
+      </View>
+    </TouchableNativeFeedback>
+  );
+};
+
+function getUpgradeInfo(state, peers): { state: UpgradeState, context: any } {
+  if (!state) {
+    return {
+      state: "Pending",
+      context: null,
+    };
+  }
+  // Error present in ANY subsystem.
+  if (state.server.state === BackendUpgradeState.Server.Error) {
+    return {
+      state: "GenericError",
+      context: state.server.context.message,
+    };
+  }
+  if (state.downloader.search.state === BackendUpgradeState.Search.Error) {
+    return {
+      state: "GenericError",
+      context: state.downloader.search.context.message,
+    };
+  }
+  if (state.downloader.download.state === BackendUpgradeState.Download.Error) {
+    return {
+      state: "GenericError",
+      context: state.downloader.download.context.message,
+    };
+  }
+  if (state.downloader.check.state === BackendUpgradeState.Check.Error) {
+    return {
+      state: "GenericError",
+      context: state.downloader.check.context.message,
+    };
+  }
+
+  // Edge case I've (kira) seen once that shouldn't happen.
+  if (state.downloader.search.state === BackendUpgradeState.Search.Idle) {
```

Comment:
I've since realized that this IS normal (I see it often), and we don't need to treat this as an error.

Reply:
I've fixed this on my latest PR fyi #580